tracer/src/Datadog.Tracer.Native/iast/hardcoded_secrets_method_analyzer.cpp

```diff
@@ -0,0 +1,81 @@
+#include "hardcoded_secrets_method_analyzer.h"
+#include "dataflow_il_rewriter.h"
+#include "iast_util.h"
+#include "method_info.h"
+#include "module_info.h"
+
+namespace iast
+{
+HardcodedSecretsMethodAnalyzer* HardcodedSecretsMethodAnalyzer::Instance = nullptr;
+
+HardcodedSecretsMethodAnalyzer::HardcodedSecretsMethodAnalyzer()
+{
+    Instance = this;
+}
+HardcodedSecretsMethodAnalyzer::~HardcodedSecretsMethodAnalyzer()
+{
+    Instance = nullptr;
+}
+
+bool HardcodedSecretsMethodAnalyzer::ProcessMethod(MethodInfo* method)
+{
+    std::vector<UserString> userStrings;
+    ILRewriter* rewriter;
+    HRESULT hr = method->GetILRewriter(&rewriter);
+    if (SUCCEEDED(hr))
+    {
+        shared::WSTRING methodName;
+        auto module = method->GetModuleInfo();
+        bool written = false;
+        for (ILInstr* pInstr = rewriter->GetILList()->m_pNext; pInstr != rewriter->GetILList();
+             pInstr = pInstr->m_pNext)
+        {
+            if (pInstr->m_opcode == CEE_LDSTR)
+            {
+                // Retrieve String
+                auto userString = module->GetUserString(pInstr->m_Arg32);
+                if (userString.size() < 10)
+                {
+                    continue;
+                }
+                if (methodName.size() == 0)
+                {
+                    methodName = method->GetFullName(false);
+                }
+                userStrings.push_back({methodName, userString});
+            }
+        }
+        hr = method->CommitILRewriter();
+    }
+
+    if (userStrings.size() > 0)
+    {
+        CSGUARD(_cs);
+        for (auto userString : userStrings)
+        {
+            _userStrings.push_back(userString);
```

Comment:
To avoid useless copies, wouldn't it make sense to use `std::move` instead ?
```
_userStrings = std::move(userString);
```

Reply:
You are right. Not very used to the move mechanics, but I agree with you.

Reply:
Oh sorry, I misread the code.
I read it again and here's my real/useful comment :) 
call `.reserve()` on `_userStrings` to avoid temporary and useless allocation (if the vector has to be resized)
Something like:
```
_userStrings.reserver(userStrings.size());
```

Reply:
That's a nice suggestion